src/Raven.Server/Documents/Subscriptions/SubscriptionConnectionState.cs

```diff
@@ -88,7 +88,10 @@ public override string ToString()
 
             var subscriptionConnection = Interlocked.CompareExchange(ref _currentConnection, incomingConnection, null);
             if (subscriptionConnection != null && subscriptionConnection != incomingConnection)
+            {
+                await Task.Delay(timeToWait);
```

Comment:
Didn't we already wait here?  `await ConnectionInUse.WaitAsync(timeToWait) `

Reply:
The problem occures when ConnectionInUse is not turned on, and we go straight to this line